### PR TITLE
TestCollect: filter incompatible marketplaces

### PR DESCRIPTION
**Pull request by dorschw**
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-3995

### CODE REVIEW

File: Tests/scripts/collect_tests/collect_tests.py [hunk 1]
```diff
@@ -304,18 +306,40 @@ def _validate_tests_in_id_set(self, tests: Iterable[str]):
...
+            reason_description: str,
+            content_item_range: Optional[VersionRange] = None,
+            allow_incompatible_marketplace: bool = False,
+    ) -> Optional[CollectionResult]:
+        pack_metadata = PACK_MANAGER.get_pack_metadata(pack_id)
+
+        try:
+            self.validate_content_item_compatibility(pack_metadata)
+        except NonXsoarSupportedPackException:
+            pass  # we do want to install packs in this case (tests are not collected in this case anyway)
```
Comment (DeanArbel) [inline Tests/scripts/collect_tests/collect_tests.py:333]: Doesnt this justify a message to the logs?

**Comment by dorschw** [comment]
nightly run https://code.pan.run/xsoar/content/-/pipelines/3600241

**Review by DeanArbel — APPROVED** [review]

File: Tests/scripts/collect_tests/collect_tests.py (hunk 1 above)
Comment (dorschw) [reply Tests/scripts/collect_tests/collect_tests.py:333]: yup

File: Tests/scripts/collect_tests/collect_tests.py [hunk 2]
```diff
@@ -257,7 +257,7 @@ def collect(self) -> Optional[CollectionResult]:
                 logger.warning('Nothing was collected, and no sanity-test-triggering files were changed')
                 return None
 
-        self._validate_tests_in_id_set(result.tests)  # type:ignore[union-attr]
+        self.__validate_tests_in_id_set(result.tests)  # type:ignore[union-attr]
```
Comment (DeanArbel) [inline Tests/scripts/collect_tests/collect_tests.py:260]: I had to read it a few times to see what was the change.

File: Tests/scripts/infrastructure_tests/tests_data/collect_tests/L_XSIAM/Packs/myXSIAMOnlyPack/Wizards/harry.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
+  "id": "wizard_harry",
```
Comment (DeanArbel) [inline Tests/scripts/infrastructure_tests/tests_data/collect_tests/L_XSIAM/Packs/myXSIAMOnlyPack/Wizards/harry.json:2]: 🧙

**Review by DeanArbel — APPROVED** [review]

**Pull request merged**